### ISSUE TITLE
Text is lost if you navigate away from a page

### ISSUE THREAD

**Issue by irilyth** [buttonmen-dev/buttonmen#2542]
If you accidentally navigate away from a page with a text box, the text is lost when you return to that page. At least in Chrome on Mac OS, this doesn't happen on other sites, e.g. GitHub, where I clicked on a link in the middle of entering this ticket, and hit the back button, and entered "mail.google.com" in the location bar and loaded Gmail, and closed the tab and did a Cmd-Shift-T to undo close tab, and in all three cases, my text in this box, and in the summary, was still there. If I do any of those things on a game page with text in the chat box, or a forum page with text in the post box, or the create page with text in the description, the text is lost when I return to the page.

This is somewhat similar to #1336 and #2296, but more general.

**Comment by irilyth** [comment]
I don't think GitHub is doing anything magical here; I see the same behavior on more or less every other site that has a box for entering text.

**Comment by blackshadowshade** [comment]
So somehow, the browser is caching text entered into text boxes? Hmmm, I've never heard of that before. Anyone more JS-savvy know about this? (Pinging @AdmiralJota)

**Comment by irilyth** [comment]
Can you reproduce it in either Chrome or your favorite browser? I don't think it's an extension or anything custom to me.

**Comment by blackshadowshade** [comment]
I can reproduce the Github behaviour on Safari. However, I'm not certain that it's not magic that Github implements.

**Comment by blackshadowshade** [comment]
https://stackoverflow.com/questions/20744644/how-to-retain-textbox-values-after-navigating-to-other-page

**Comment by AdmiralJota** [comment]
My strong guess is that this is because we build the contents of every page dynamically in javascript. Normal sites render the page layout on the server when the page is loaded. We render an empty page on the server and run javascript on the client that builds everything laboriously by hand. That means that when the browser gets the page from the server, there _is_ no text box for it to put that cached text into.

**Comment by irilyth** [comment]
Yeah, that's from like six years ago. I think Chrome (and perhaps Safari) just do this by default nowadays. Try some other sites too?
